### DISCUSSION TITLE
Missing values when the VM is shutdown for a long period

### DISCUSSION THREAD

**Discussion by liranr23** [vmware/govmomi#3401] [Community Help]
Hi all,
We are using `govmomi v0.34.1` to gather data of vSphere instance.
For example, we use it to gather the operation system a VM is set with `summary.guest.guestId`.
Recently, we saw in a large vSphere instance that this field is missing when getting it (as far as I know it's a static field of the VM, we also saw it on `summary.guest.guestId` to get the static IPs of the guest - but that's dynamic).
When we boot the VM and shut it, then we can get those details.
Regardless, even without booting the guest, when creating an OVF for example, the operation system is set.
I wonder, is that the intended behavior?

**Comment by dougm** [comment]
[VirtualMachineGuestSummary](https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vim.vm.Summary.GuestSummary.html) and [GuestInfo](https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vim.vm.GuestInfo.html) depend on vmware-tools running inside the VM to report this data. When tools stops reporting after some time, vCenter will consider the guest data stale and returns empty values to the API calls (which you should also see in the UI).

**Reply by liranr23** [reply]
Thanks Doug!
